src/document/generator.ml

```diff
@@ -646,6 +646,11 @@ module Make (Syntax : SYNTAX) = struct
         O.keyword "type" ++ O.txt " "
         ++ Link.from_path (t.type_path :> Paths.Path.t)
         ++ O.txt " +=" ++ O.sp
+        ++
+        if t.private_ then
+          O.keyword Syntax.Type.private_keyword
+          ++ match t.constructors with [] -> O.noop | _ :: _ -> O.sp
```

Comment:
I think that extensions never have an empty list of new constructor, so it's safe and more readable to write:

```suggestion
          O.keyword Syntax.Type.private_keyword ++ O.sp
```
(I guess ocamlformat would format it this way)